quickwit/quickwit-indexing/src/source/gcp_pubsub_source.rs

```diff
@@ -440,11 +515,182 @@ mod gcp_pubsub_emulator_tests {
             "index_id": index_id,
             "source_id": source_id,
             "subscription": subscription,
+            "partition_id": partition,
             "num_bytes_processed": 54,
             "num_messages_processed": 6,
             "num_invalid_messages": 0,
             "num_consecutive_empty_batches": 4,
         });
         assert_eq!(exit_state, expected_exit_state);
     }
+
+    #[tokio::test]
+    async fn test_gcp_pubsub_source_ack() {
```

Comment:
We could drop the line `self.subscription.ack(message_ids).await`, and this test would still pass. Does the PubSub API have some way to tell us that we acknowledged the messages correctly?

Reply:
Yeah... sadly I think there is no way to really know if the message are ack. 
My guess is because gcp doesn't garantee you that the message will not be redeliver even when you call ack.

When I was working with go, we usually tested that with an interface mock...

## Options1:
One way, would be to set an ack deadline really small. And at the end of the test just subscribe and wait to see if we receive message. It will not be perfect but it should be "fine". Problem is we might need to wait few seconds... 

## Options2:

having some interface so we can mock / ensure that the function was call.

Reply:
I'm leaning towards Option 2. We can only mock the call to `ack` for now. There are examples in the codebase of mocked traits (`Metastore`, code-generated clients) with [`mockall`](https://docs.rs/mockall/latest/mockall/).